### PR TITLE
Analytics: Add setting to disable tracking for logged-in users

### PR DESCRIPTION
**Pull request by aaemnnosttv**
## Summary

Addresses issue #88

## Relevant technical choices

* Adds a new `GET tracking-disabled` datapoint for consistency with the other settings-based datapoints in the module
* Adds a new `googlesitekit_analytics_tracking_disabled` filter for developers to have full control over snippet output
* Designed in a way which should be easy to extend in the future with other options for exclusions

## Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.4.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: tests/phpunit/integration/Modules/AnalyticsTest.php [hunk 1]
```diff
@@ -226,6 +226,8 @@ public function test_tracking_disabled( $settings, $logged_in, $test_method ) {
 		wp_scripts()->queue      = array();
 		wp_scripts()->done       = array();
 		remove_all_actions( 'wp_enqueue_scripts' );
+		// Remove irrelevant script from throwing errors in CI from readfile().
+		remove_action( 'wp_head', 'print_emoji_detection_script', 7 );
```
Comment (tofumatt) [inline tests/phpunit/integration/Modules/AnalyticsTest.php]: This is amazing. Nice find 🔍

File: includes/Modules/Analytics.php [hunk 2]
```diff
@@ -159,6 +166,25 @@ function( $data ) {
 		);
 	}
 
+	/**
+	 * Checks whether or not tracking snippet should be contextually disabled for this request.
+	 *
+	 * @return bool
```
Comment (felixarntz) [inline includes/Modules/Analytics.php]: This needs a `@since` annotation.

Comment (aaemnnosttv) [reply includes/Modules/Analytics.php]: Do all methods need this or only public? (I thought the latter)

Comment (felixarntz) [reply includes/Modules/Analytics.php]: We should have this on all methods.

**Comment by aaemnnosttv** [comment]
> Revisiting this, can we update the design so that it looks consistent with the other existing text?

Sure. I thought this should be different as the current text is simply a paragraph, not really styled as a label/heading.

**Comment by aaemnnosttv** [comment]
Updated!

Edit mode
image

Display mode
image

**Comment by boryn** [comment]
But still, how to check inside the GTM that the user is logged in? So far I check if the `#wpadminbar` div exists. It would be great if logged in status as well as user role were passed.

**Comment by Soetens** [comment]
The addition of `googlesitekit_analytics_tracking_disabled` is great, but should also be added to Tag Manager.

**Comment by aaemnnosttv** [comment]
@Soetens at the time of this PR, the Tag Manager module required Analytics to be set up and configured first so it wasn't necessary, but I agree that this makes more sense now that the modules do not depend on each other.

**Pull request merged**